### DISCUSSION TITLE
What should happen to the fits when starting a new ODMR measurement?

### DISCUSSION THREAD

**Discussion by simonschmitt2** [Ulm-IQO/qudi-iqo-modules#137] [Q&A] [answered]
Currently, when starting an ODMR measurement, any fits from the previous measurement remain visible. This can be especially annoying when the scan range has changed (see screenshot below). In the `start_odmr_scan` method within the `odmr_logic` file, there is a line `# ToDo: Clear old fit`, suggesting that the behavior described above is not the expected behavior. I am happy to implement the missing functionality, but I am not exactly sure what the expected behavior should be since I am not familiar with the `Data Channel` and `Range Index` parameters in the GUI. Should all fits be removed for all ranges, or is the requirement more complex?

![screenshot2](https://github.com/Ulm-IQO/qudi-iqo-modules/assets/18697822/5ab84730-f9de-4e39-b60f-6b6eb0121137)

**Comment by NicolasStaudenmaier** [comment, marked as answer]
Thanks Simon for pointing at this problem. As discussed just now in the meeting the behavior should be that with a new scan all old data will be erased. This includes the fits as well. So it makes sense to clear the fits of all ranges and all channels.
